solvers/cyanure.py

```diff
@@ -14,13 +15,16 @@ class Solver(BaseSolver):
 
     def set_objective(self, X, y, lmbd):
         self.X, self.y, self.lmbd = X, y, lmbd
+        if scipy.sparse.issparse(self.X) and \
+           scipy.sparse.isspmatrix_csc(self.X):
+            self.X = scipy.sparse.csr_matrix(self.X)
 
         n_samples = self.X.shape[0]
 
         self.solver = Regression(loss='square', penalty='l1',
                                  fit_intercept=False)
         self.solver_parameter = dict(
-            lambd=self.lmbd / n_samples, solver='auto',
+            lambd=self.lmbd / n_samples, solver='auto', it0=1000000,
```

Comment:
Just out of curiosity, what does this parameter controls?

Reply:
The code computes duality gaps + evaluate the objective every it0 iterations. In the context of this benchmark, there is no need to do it since the amount of computation is controlled by the number of iterations.

Reply:
Indeed! thx